### ISSUE TITLE
Connection/Authentication issue

### ISSUE THREAD

**Issue by Zaista** [Qytera-Gmbh/cypress-xray-plugin#69]
I'm stuck figuring out what could be causing this error, and I'm at the point of giving up. This only manifests on my CI/CD pipeline while it works locally, the error message from `importExecutionResultsError.json` file is:

```
{"error":{"message":"read ECONNRESET","name":"Error","stack":"Error: read ECONNRESET
   at Function.AxiosError.from (/e2e/node_modules/axios/lib/core/AxiosError.js:89:14)
   at RedirectableRequest.handleRequestError (/e2e/node_modules/axios/lib/adapters/http.js:577:25)
   at RedirectableRequest.emit (node:events:513:28)
   at RedirectableRequest.emit (node:domain:489:12)
   at ClientRequest.eventHandlers.<computed> (/e2e/node_modules/follow-redirects/index.js:14:24)
   at ClientRequest.emit (node:events:513:28)
   at ClientRequest.emit (node:domain:489:12)
   at Socket.socketErrorListener (node:_http_client:502:9)
   at Socket.emit (node:events:513:28)
   at Socket.emit (node:domain:489:12)
   at emitErrorNT (node:internal/streams/destroy:151:8)
   at emitErrorCloseNT (node:internal/streams/destroy:116:3)
   at processTicksAndRejections
```

System info:
TeamCity as build server
Jira DataCenter + Xray plugin

I've setup authentication using `JIRA_API_TOKEN` and don't think it's related to this as locally it works.
Plus, don't think it's connection issues between TeamCity and Jira, because another solution that uses node-fetch package works without issues...

Any ideas, thoughts are appreciated

**Comment by csvtuda** [comment]
Hey @Zaista,

I don't really know what the error might be based on the short snippet. The error seems to be appearing on executions results import, correct? Authentication works fine? You should see corresponding log messages, e.g. `Authentication successful.`.

A few things you could try out:
- Set `plugin.debug` to `true` in the [configuration](https://qytera-gmbh.github.io/projects/cypress-xray-plugin/section/configuration/plugin/#debug). This will write all request and response payloads the plugin sends to Jira/Xray and receives in return to individual JSON files. You could then download them from your CI/CD setup (hopefully - if not you could simply echo their content in the worst case) and inspect them for further analysis.
- How long does an upload take when uploading from your local machine? Maybe the request is too big and the connection closes somewhere in the middle of the upload? You could combine this with the `debug` option and simply inspect the payload file sizes.

**Comment by Zaista** [comment]
Hi @csvtuda 

The first thing I did was to mock the simple `axios` request from my local machine, and that does indeed work.

The Authentication part also doesn't work, if you mean the stuff that happens in `beforeHook` in the plugin. The error is actually what I shared: 
Cypress Xray Plugin | ERROR | Failed to upload results to Xray: "Error: read ECONNRESET" 

and when importing part comes it just hangs there waiting for the Jira to respond:
| Cypress Xray Plugin | INFO | Waiting for https://jira.cloudflight.io/ to respond... (310 seconds)
after which it fails.

I'll try to experiment with debug option, but it's not straightforward getting those error files from the build agents. Anyway, I'll look it up

**Comment by csvtuda** [comment]
Thanks for the feedback.

I have two small suggestions:

- Try to update to the latest version, which is `3.3.0`. Version `3.2.0` fixed a bug which kept printing these messages without ever stopping, even when a response had been received a long time ago:
  ```
  | Cypress Xray Plugin | INFO | Waiting for https://jira.somewhe.re/ to respond... (290 seconds)
  | Cypress Xray Plugin | INFO | Waiting for https://jira.somewhe.re/ to respond... (300 seconds)
  | Cypress Xray Plugin | INFO | Waiting for https://jira.somewhe.re/ to respond... (310 seconds)
  ```
  It does not really affect your problem (I assume), but it should clarify what the plugin is really doing (i.e. nothing in this case).

- Regarding the download of the debug payloads: If it's really difficult, would it be possible for you to commit a shell/batch script containing your hardcoded, mimicked request? Just tell the pipeline to execute that script then instead.

**Comment by csvtuda** [comment]
Hey @Zaista,

I encountered this error today as well (only once though). It was a very small payload which resulted in an instantaneous upload usually.

However, I was trying to upload from inside a VPN. Is your Xray instance or your TeamCity build server inside a VPN by chance? Maybe it's causing a few connection hiccups here and there.

**Comment by Zaista** [comment]
Ok, so currently the plugin sorta works :D 
I update the plugin and cypress version to latest ones, and it seems to be working now, things get uploaded to xray.

It's just that, each spec creates a separate execution, is this known perhaps? 🤔 
I'll investigate further, maybe you can even close this issue, and I can open another one if the problem persists...

**Comment by csvtuda** [comment]
I'm glad to hear that it's sort of working.

Regarding the creation of individual *test execution issues* for each spec, that should definitely *not* happen. The plugin should *at most* create the following issues, unless you've set up issue reuse as described below:
- One *test issue* for each test, i.e. each `it(...)`.
- One *test execution issue* containing all corresponding test issues.

Just in case, you should be able to reuse:
- existing *test plan issues* through the [`testPlanIssueKey` option](https://qytera-gmbh.github.io/projects/cypress-xray-plugin/section/configuration/jira/#testplanissuekey)
- existing *test execution issues* through the [`testExecutionIssueKey` option](https://qytera-gmbh.github.io/projects/cypress-xray-plugin/section/configuration/jira/#testexecutionissuekey)
- existing *test issues* through inclusion of their issue keys in the test titles, as described [here](https://qytera-gmbh.github.io/projects/cypress-xray-plugin/section/guides/targetingExistingIssues/#reuse-test-case-issues)

I will close the issue as you suggested. I feel like it's more of a networking issue, not a plugin issue. But feel free to open a new issue if this problem or your separate execution issues persist!
